core/src/main/java/com/microsoft/applicationinsights/TelemetryClient.java

```diff
@@ -217,42 +217,66 @@ public void trackTrace(TraceTelemetry telemetry) {
+     * @deprecated Use {@link #trackMetric(String, double, Integer, Double, Double, Double, Map)}
      */
+    @Deprecated
     public void trackMetric(String name, double value, int sampleCount, double min, double max, Map<String, String> properties) {
+        this.trackMetric(name, value, sampleCount, min, max, null, properties);
+    }
+
+    /**
+     * Sends a numeric metric to Application Insights. Appears in customMetrics in Analytics, and under Custom Metrics in Metric Explorer.
+     *
+     * @param name The name of the metric. Max length 150.
+     * @param value The value of the metric. Sum if it represents an aggregation.
+     * @param sampleCount The sample count.
+     * @param min The minimum value of the sample.
+     * @param max The maximum value of the sample.
+     * @param stdDev The standard deviation of the sample.
+     * @param properties Named string values you can use to search and classify trace messages.
+     * @throws IllegalArgumentException if name is null or empty
+     */
+    public void trackMetric(String name, double value, Integer sampleCount, Double min, Double max, Double stdDev, Map<String, String> properties) {
         if (isDisabled()) {
             return;
         }
 
-        if (Strings.isNullOrEmpty(name)) {
-            name = "";
-        }
-
         MetricTelemetry mt = new MetricTelemetry(name, value);
         mt.setCount(sampleCount);
-        if (sampleCount > 1) {
-            mt.setMin(min);
-            mt.setMax(max);
-        }
-
+        mt.setMin(min);
+        mt.setMax(max);
+        mt.setStandardDeviation(stdDev);
         if (properties != null && properties.size() > 0) {
-            MapUtil.copy(properties, mt.getContext().getProperties());
+            MapUtil.copy(properties, mt.getProperties());
         }
-
         this.track(mt);
     }
 
     /**
      * Sends a numeric metric to Application Insights. Appears in customMetrics in Analytics, and under Custom Metrics in Metric Explorer.
      * @param name The name of the metric. Max length 150.
-     * @param value The value of the metric. Should be greater than 0.
+     * @param value The value of the metric.
+     * @throws IllegalArgumentException if name is null or empty.
      */
     public void trackMetric(String name, double value) {
-        trackMetric(name, value, 1, value, value, null);
+        trackMetric(name, value, null, null, null, null, null);
+    }
+
+    /**
+     * Sends a numeric metric to Application Insights. Appears in customMetrics in Analytics, and under Custom Metrics in Metric Explorer.
+     * @param name The name of the metric. Max length 150.
+     * @param value The value of the metric.
+     * @param properties Named string values you can use to search and classify trace messages.
+     * @throws IllegalArgumentException if name is null or empty.
+     */
+    public void trackMetric(String name, double value, Map<String, String> properties) {
```

Comment:
Is this the new public API we want to expose? The method looks pretty handy but we should just make sure, that other SDKs also have it. Even if they don't I am fine with introducing this method, just good to check and suggest them too. I was looking for this method too and was surprised didn't existed!

Reply:
https://github.com/Microsoft/ApplicationInsights-dotnet/blob/39a5ef23d834777eefdd72149de705a016eb06b0/src/Microsoft.ApplicationInsights/TelemetryClient.cs#L228

Reply:
Cool makes sense.

Reply:
Let's please not make any more overloads that take non-aggregates (like this one).
In the mid-term we want to deprecate the existing ones too :)